core/src/main/scala/com/softwaremill/session/SessionManager.scala

```diff
@@ -84,7 +84,19 @@ trait CsrfManager[T] {
   def config: SessionConfig
 
   def tokenInvalidRejection = AuthorizationFailedRejection
-  def createToken(): String = SessionUtil.randomString(64)
+  def createToken(): String = {
+    val timestamp = System.currentTimeMillis().toString
+    val hmac = generateHmac(timestamp)
+    s"$hmac$timestamp"
+  }
+  def validateToken(token: String): Boolean = {
+    val len = System.currentTimeMillis().toString.length
```

Comment:
I suppose we are assuming that the timestamp will always be `13` digits? Or rather, that between invocations of `create` and `validate` no extra digit will be added - unlikely, but possible ;). If we make this assumption, I think I would document it & add a constant, or use some other format which allows separating timestamp & hmac is a predictable way

Reply:
Added a specific encoder/decoder to improve this section.